### ISSUE TITLE
MD024/no-duplicate-heading regression in 0.11.0

### ISSUE THREAD

**Issue by mvystrcil** [DavidAnson/markdownlint-cli2#240]
Configuration file `.markdownlint-cli2.jsonc` with 
```
		//MD024
		"no-duplicate-header": {
			"allow_different_nesting": true
		},
```

When using version `0.10.0`, no errors are found.

When using new version `0.11.0`, errors with following are found `docs/diagnostika_a_statistiky/sp_monitoring_a_statistiky/konfigurace.md:49 MD024/no-duplicate-heading Multiple headings with the same content [Context: "##### COM port"]`

I don't expect these errors to be found, when `MD024` should be ignored.

**Comment by DavidAnson** [comment]
The “header” aliases for rules like `MD024/no-duplicate-heading` have been deprecated since v0.9.0 of the `markdownlint` library released in May of 2018. (The change was made to fix <https://github.com/DavidAnson/markdownlint/issues/113> which pointed out the correct term is “heading”.) Version v0.32.0 of the `markdownlint` library removed the deprecated “header” aliases. Therefore, rules like this need to referred to by their “heading” alias or by rule name. Please update your configuration to resolve the issue. Thank you!

**Comment by mvystrcil** [comment]
@DavidAnson thank you for explanation. Would it be possible to implement some sort of configuration validation ? I mean, if linter would told me, this configuration item is deprecated, I would try to search for reason why.

**Comment by DavidAnson** [comment]
I don't know how big of a problem this will be yet. I have a few ideas if it becomes prominent.

FYI, there is already a schema that can be used to validate configuration: <https://github.com/DavidAnson/markdownlint-cli2/tree/main/schema>. And it is applied automatically when using the VS Code extension for markdownlint.
